reconcile/blackbox_exporter_endpoint_monitoring.py

```diff
@@ -0,0 +1,135 @@
+import logging
+import sys
+from urllib.parse import urlparse
+from typing import Optional
+from dataclasses import asdict
+
+from reconcile import queries
+from reconcile.models import Endpoint, EndpointMonitoringProvider
+from reconcile.utils.openshift_resource import (
+    OpenshiftResource, ResourceInventory)
+from reconcile.utils.defer import defer
+from reconcile.utils.semver_helper import make_semver
+import reconcile.openshift_base as ob
+
+QONTRACT_INTEGRATION = "blackbox-exporter-endpoint-monitoring"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+LOG = logging.getLogger(__name__)
+
+
+def parse_prober_url(url: str) -> dict[str, str]:
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in ["http", "https"]:
+        raise ValueError(
+            "the prober URL needs to be an http:// or https:// one "
+            f"but is {url}"
+        )
+    data = {
+        "url": parsed_url.netloc,
+        "scheme": parsed_url.scheme
+    }
+    if parsed_url.path:
+        data["path"] = parsed_url.path
+    return data
+
+
+def build_probe(provider: EndpointMonitoringProvider,
+                endpoints: list[Endpoint]) -> Optional[OpenshiftResource]:
+    blackbox_exporter = provider.blackboxExporter
+    if blackbox_exporter:
+        prober_url = parse_prober_url(blackbox_exporter.exporterUrl)
+        body = {
+            "apiVersion": "monitoring.coreos.com/v1",
+            "kind": "Probe",
+            "metadata": {
+                "name": f"endpoint-monitoring-{provider.name}",
+                "namespace": blackbox_exporter.namespace.name,
+                "labels": {
+                    "prometheus": "app-sre"
```

Comment:
probe labels (prometheus selector) should not be hard coded. this is too specific for AppSRE